src/legacy/ui/ui_render/ui_render_mixin.js

```diff
@@ -198,16 +203,15 @@ export function uiRenderMixin(kbnServer, server, config) {
       basePath: request.getBasePath(),
       serverName: config.get('server.name'),
       devMode: config.get('env.dev'),
-      uiSettings: await props({
-        defaults: uiSettings.getDefaults(),
-        user: includeUserProvidedConfig && uiSettings.getUserProvided()
-      })
+      uiSettings: await getUiSettings({ request, includeUserProvidedConfig }),
     };
   }
 
   async function renderApp({ app, h, includeUserProvidedConfig = true, injectedVarsOverrides = {} }) {
     const request = h.request;
     const basePath = request.getBasePath();
+    const uiSettings = await getUiSettings({ request, includeUserProvidedConfig });
+    app = app || { getId: () => 'core' };
```

Comment:
shouldn't we cover it with unit/integration tests?

Reply:
Good call, I'll add some tests for this file.